File: mgmt-back/app/lib/utils/order.py
from app.database.models import Order, OrderItem
from app.database.mongo import DBClient
def get_dish_cost(dish_id: str, db_client: DBClient):
    dish = db_client.get_dish_by_id(dish_id)
    return dish.cost_unit if dish else 0

def order_from_items(items: list[OrderItem], db_client: DBClient) -> Order:
    total_cost = sum(
        item.quantity * (
            get_dish_cost(item.id_dish, db_client) +
            sum(extra.cost for extra in (item.selected_extras or []))
        )
        for item in items
    )
    return Order(
        _id=None,
        id_user="",
        items=items,
        total_cost=total_cost,
        status="preparing",
        type="dinein"
    )

File: register/source/lib/types/order.ts
import type { DishExtra } from "./dish.js";

/**
 * 
 * class OrderItem(BaseModel):
    id_dish: str
    quantity: int
    selected_extras: Optional[List[DishExtra]] = None


OrderStatus = Literal["preparing", "done", "delivered"]
OrderType = Literal["dinein", "delivery"]

class Order(BaseModel):
    id: Optional[str] = Field(None, alias="_id")
    id_user: str
    items: List[OrderItem]
    total_cost: float
    status: OrderStatus
    type: OrderType
    created_at: datetime = Field(default_factory=datetime.now)
    updated_at: Optional[datetime] = None
   
 */

export type OrderStatus = 'preparing' | 'done' | 'delivered';
export type OrderType = 'dinein' | 'delivery';

export interface OrderItem {
  id_dish: string;
  quantity: number;
  selected_extras?: DishExtra[];
}

export interface Order {
  _id?: string;
  id_user: string;
  items: OrderItem[];
  total_cost: number;
  status: OrderStatus;
  type: OrderType;
  created_at: string;
  updated_at?: string;
}

